### PR TITLE
MOSIP-17075

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			anonymousProfileRepository.save(anonymousProfileEntity);
+			} catch (DataAccessLayerException e) {
+				regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+						"", e.getMessage() + ExceptionUtils.getStackTrace(e));
+
+				throw new TablenotAccessibleException(
+						PlatformErrorMessages.RPR_RGS_ANONYMOUS_PROFILE_TABLE_NOT_ACCESSIBLE.getMessage(), e);
+			} 
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), regId,
+					"PacketInfoManagerImpl::saveRegLostUinDetData()::exit");
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java]: update the logger correctly saveRegLostUinDetData..

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java [hunk 2]
```diff
@@ -520,16 +527,36 @@ private boolean validateRegistrationID(SyncRegistrationDto registrationDto,
...
+		AnonymousProfileDTO dto=new AnonymousProfileDTO();
+		try{
+			dto.setProcessName(registrationDto.getRegistrationType());
+			dto.setStatus("REGISTERED");
+			dto.setStartDateTime(timeStamp);
+			dto.setDate(LocalDate.now(ZoneId.of("UTC")).toString());
+			dto.setProcessStage("SYNC");
+			dto.setEmail(dto.getEmail() != null ? getHashCode(dto.getEmail()) : null);
+			dto.setEnrollmentCenterId(referenceId.split("_")[0]);
+			dto.setPhone(dto.getPhone() != null ? getHashCode(dto.getPhone()) : null);
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java]: please dont set phone or email actual value. Its like this -
1. if phone is there then from mapping json read phone field. Update the field here. If phone number is not there then update as null
2. same for email.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	@Override
+	public void saveAnonymousProfile(String regId, String processStage, String profileJson) {
+		
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), regId,
+					"PacketInfoManagerImpl::saveAnonymousProfile()::entry");
+			try {
+			AnonymousProfileEntity anonymousProfileEntity=new AnonymousProfileEntity();
+			AnonymousProfilePKEntity anonymousProfilePKEntity=new AnonymousProfilePKEntity();
+			anonymousProfilePKEntity.setId(regId);
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java]: id should not be regid. its random uuid. use -
RegistrationUtility.generateId() no need to pass regid in this method

**Review by MonobikashDas — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java (hunk 1 above)
Comment (ghost) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java]: I have changed the code.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java (hunk 3 above)
Comment (ghost) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java]: I have changed the code.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java (hunk 2 above)
Comment (ghost) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java]: I have changed the code

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java [hunk 4]
```diff
@@ -520,16 +537,47 @@ private boolean validateRegistrationID(SyncRegistrationDto registrationDto,
...
+	private void saveAnonymousProfile(SyncRegistrationDto registrationDto, String referenceId, String timeStamp)  {
+		AnonymousProfileDTO dto=new AnonymousProfileDTO();
+		try{
+			dto.setProcessName(registrationDto.getRegistrationType());
+			dto.setStatus("REGISTERED");
+			dto.setStartDateTime(timeStamp);
+			dto.setDate(LocalDate.now(ZoneId.of("UTC")).toString());
+			dto.setProcessStage("SYNC");
+			List<String> channel=new ArrayList<>();
+			RestTemplate restTemplate = new RestTemplate();
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java]: use below code to get mapping json-
JSONObject regProcessorIdentityJson = utility.getRegistrationProcessorMappingJson(MappingJsonConstants.IDENTITY);
JsonUtil.getJSONObject(regProcessorIdentityJson, MappingJsonConstants.EMAIL), MappingJsonConstants.VALUE)

**Review by MonobikashDas — APPROVED** [review]

**Pull request merged**